src/core/database/database.cpp

```diff
@@ -0,0 +1,31 @@
+// SPDX-License-Identifier: MIT
+// SPDX-FileCopyrightText: 2024 NeoFOAM authors
+
+#include <stdexcept>
+
+#include "NeoFOAM/core/database/database.hpp"
+#include "NeoFOAM/core/database/collection.hpp"
+
+namespace NeoFOAM
+{
+
+Collection& Database::insert(const std::string& name, const Collection& col)
+{
+    collections_.emplace(name, col);
+    return collections_.at(name);
```

Comment:
```suggestion
    return *(collections_.emplace(name, col).first);
```
you don't need a range check access and [emplace](https://en.cppreference.com/w/cpp/container/unordered_map/emplace) returns an iterator for an `unordered_map`.